### ISSUE TITLE
JSImported modules are eagerly required

### ISSUE THREAD

**Issue by julienrf** [scala-js/scala-js#2711]
In vanilla JavaScript you can do that:

~~~ javascript
window.setTimeout(function () {
  require('launch-missile');
}, 1000);
~~~

Note that the effect of requiring a module is delayed.

With Scala.js, the equivalent would be:

~~~ scala
@JSImport("launch-missile", Namespace)
@js.native
object LaunchMissile extends js.Any

js.timers.setTimeout(1000)(LaunchMissile)
~~~

However, the code generated by Scala.js eagerly `require`s the `launch-missile` module.

Would it be possible to effectively load the module at the time it is actually used?

The motivation is that some existing NPM modules have side-effects at load time.

**Comment by sjrd** [comment]
> Would it be possible to effectively load the module at the time it is actually used?

No, that's not possible, because our semantics are ES6 semantics. ES6 does not permit lazy imports, so our semantics do not permit it either. If we changed our semantics, we would never be able to emit ES6 modules in the future.

Hopefully JavaScript libraries will stop using such side-effects at load time when converting to ES6 modules themselves.
